### PR TITLE
Add .jvmopts and switch to setup-java action

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

File: .jvmopts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+-Xss4m
+-Xms1G
+-Xmx2G
+-XX:ReservedCodeCacheSize=1024m
+-XX:+TieredCompilation
+-Dfile.encoding=UTF-8
```
Comment (adpi2) [inline .jvmopts:6]: What's the purpose of those options?

Comment (tgodzik) [reply .jvmopts:6]: just copied over the ones in Metals, they seem to work fine here also, but I can pinpoint exactly which are needed in this case.

It seems though there is an issue with the Windows build anyway, will need to take a look at that

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -9,23 +9,23 @@ jobs:
...
     steps:
     - uses: actions/checkout@v2.3.4
-    - uses: olafurpg/setup-scala@v10
+    - uses: actions/setup-java@v3
       with:
+        distribution: 'temurin'
         java-version: ${{ matrix.jdk }}
     - name: Cross-publish the expression evaluator
-      run: sbt --client + expressionCompiler/publishLocal
+      run: sbt "--client +expressionCompiler/publishLocal"
```
Comment (tgodzik) [inline .github/workflows/ci.yml:28]: It seems that I am unable to make `--client` work on windows :thinking: 

Not sure why that is in this case

**Comment by ckipp01** [comment]
It looks like this has been switched to use the coursier setup action. So I'll go ahead and close this.

**Pull request closed without merging**